app/controllers/spree/admin/reports_controller_decorator.rb

```diff
@@ -181,20 +189,37 @@ def sales_tax
   end
 
   def bulk_coop
-    # -- Prepare date parameters
-    prepare_date_params params
+    if request.format.json?
```

Comment:
Is there any reason we're not creating or using a pure JSON API here? From what I understand this controller was meant to serve HTML, right?

If the issue is reusing the logic, maybe it's a good time to move it into some business logic class. Like a service or something like that. Thoughts?